core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java

```diff
@@ -270,7 +274,10 @@ public SparkSqlDialect(SqlDialect.Context context) {
         writer.endFunCall(lengthFrame);
         break;
       case EXTRACT:
-        final SqlWriter.Frame extractFrame = writer.startFunCall(call.operand(0).toString());
+        String extractDateTimeUnit = call.operand(0).toString();
+        String resolvedDateTimeFunctionName = extractDateTimeUnit.equalsIgnoreCase("WEEK")
```

Comment:
Can you declare these constants in `DateTimestampFormatUtil` and use it from there?

Reply:
done